SpotGamma.cpp

```diff
@@ -1,249 +1,284 @@
+        if (sc.HTTPResponse != "") {
+            RequestState = REQUEST_RECEIVED;
+			sc.AddMessageToLog("HTTP Request RECEIVED, Updating Levels", 1);
+            // sc.AddMessageToLog(sc.HTTPResponse, 1); // HTTP response for debugging
+            HttpResponseContent = sc.HTTPResponse;
         } else {
-          colorString += "CCCCCC";
+        // Request not completed, nothing to do
+			// sc.AddMessageToLog("Hanging Request", 0);
+        return;
         }
-      }
-
-      index++;
     }
 
-    int r, g, b;
-    sscanf_s(colorString.GetChars(), "%02x%02x%02x", &r, &g, &b);
-    const int color = colorOverride != RGB(0, 0, 0) ? colorOverride : RGB(r, g, b);
+    // reset state for next run
+    RequestState = REQUEST_NOT_SENT;
+	// StateMessage.Format("sc.Index: %d, UpdateIntervalInMin: %d, RequestState: %d", sc.Index, UpdateIntervalInMin, RequestState);
+	// sc.AddMessageToLog(StateMessage, 0);
+    // sc.AddMessageToLog("RequestState Reset: REQUEST_NOT_SENT", 0);
 
-    if (!(priceLevel > 0)) {
-      return;
-    }
+    // Tokenize
+    // Initialize vector to store tokens from CSV
+    std::vector<char*> tokens;
+    // Create input stream from CSV content retrieved from HTTP Response
+    std::istringstream input(HttpResponseContent.GetChars());
+    int LineNumber = 1;
+
+    // We're going to:
+    // - loop line by line
+    // - convert each line to SCString for processing
+    // - remove " characters as first and last characters which cause an error
+    // - tokenize the line 
+    // - initialize the Tool and tool settings
+    // - then process each token extracted from the current line
+    for (std::string line; getline(input,line);) {
+        // msg.Format("%s", line.c_str());
+        // sc.AddMessageToLog(msg,1);
+
+        SCString scline = line.c_str();
+		// May be a Google Sheets specific issue, unsure
+        scline = scline.GetSubString(scline.GetLength() - 2, 1);
+        scline.Tokenize("\",\"", tokens);
+
+        s_UseTool Tool;
+        Tool.LineStyle = LINESTYLE_SOLID;
+        Tool.LineWidth = 1;
+        Tool.TextAlignment = DT_RIGHT;
+        int TokenIndex = 1;
+        float price;
+        SCString note;
+        SCString color;
+        int linewidth = 1;
+        int textalignment = 1;
+        for (char* i : tokens) {
+            //msg.Format("%s", i);
+            //sc.AddMessageToLog(msg,1);
+
+            // price
+            if (TokenIndex == 1) {
+                // Should technically be replaced with stof(i) for type checking
+                // we control the data so "assume" type is proper
+                price = atof(i);
+                Tool.DrawingType = DRAWING_HORIZONTALLINE;
+                Tool.BeginValue = price;
+                Tool.EndValue = price;
+            }
+
+            // note
+            else if (TokenIndex == 2) {
+                note = i;
+            }
+            // color
+            else if (TokenIndex == 3) {
+                color = i;
```

Comment:
I'm confused.  We're setting `color` to something based on the CSV payload?  I don't think we send that down anywhere...